impl/src/main/java/org/jboss/weld/util/bytecode/ClassFileUtils.java

```diff
@@ -45,6 +45,8 @@ private ClassFileUtils() {
 
     static {
         try {
+            // TODO this needs to go away anyway, else we would blow up on JDK 12 even if we have the new approach
+            // probably revert to setAccessible() for some later version?
```

Comment:
I still need to think how to handle this static block (which is what causes JDK 12 exceptions) and if we want to address it now.

We could at least move this to a separate method (which we would need to thread-safely invoke only once) that wouldn't get invoked if we know we can use the other approach. That way, we would at least avoid crashing if the integrator did their part.

Reply:
> I still need to think how to handle this static block (which is what causes JDK 12 exceptions)

What's exactly wrong with static initializer in JDK12?

Reply:
Not with initializer, with the following line that we invoke:
`theUnsafe.objectFieldOffset(AccessibleObject.class.getDeclaredField("override"));`

Having it in initializer means it will get called every time, even should we use the new API and don't need this at all.

Reply:
Ok, then we could probably use static field of type [LazyValueHolder](https://github.com/weld/core/blob/master/impl/src/main/java/org/jboss/weld/util/LazyValueHolder.java) and call `getDeclaredMethod()` in its `computeValue()`?